tests/test_subparser.py

```diff
@@ -0,0 +1,263 @@
+from dataclasses import dataclass, field
+from typing import Union
+from sys import version_info
+
+from pytest import mark, raises
+
+from pydargs import parse
+
+
+@dataclass
+class Action1:
+    a: int
+    b: str = "abc"
+
+
+@dataclass
+class Action2:
+    c: int = 42
+    b: str = "abc"
+    e: str = field(default="def", metadata=dict(positional=True))
+
+
+@dataclass
+class Action3:
+    d: float
+    e: list[str] = field(default_factory=lambda: ["def"])
+
+
+@dataclass
+class Action4:
+    sub_action: Union[Action1, Action2]
```

Comment:
Maybe a bit cumbersome, but can we have some SubAction1 and SubAction2 here, to make this particular test a little easier to comprehend